src/server/src/main/java/io/cassandrareaper/storage/MemoryStorageFacade.java

```diff
@@ -90,37 +105,48 @@ public final class MemoryStorageFacade implements IStorageDao {
   private final MemorySnapshotDao memSnapshotDao = new MemorySnapshotDao();
   private final MemoryMetricsDao memMetricsDao = new MemoryMetricsDao();
   private final ReplicaLockManagerWithTtl replicaLockManagerWithTtl;
+  private final String persistenceStoragePath;
 
   public MemoryStorageFacade(String persistenceStoragePath, long leadTime) {
     LOG.info("Using memory storage backend. Persistence storage path: {}", persistenceStoragePath);
-    this.embeddedStorage = EmbeddedStorage.Foundation(Paths.get(persistenceStoragePath))
-        .onConnectionFoundation(
-            c -> {
-              c.setFieldEvaluatorPersistable(TRANSIENT_FIELD_EVALUATOR);
-            }
-        ).createEmbeddedStorageManager();
-    this.embeddedStorage.start();
-    if (this.embeddedStorage.root() == null) {
-      LOG.info("Creating new data storage");
-      this.memoryStorageRoot = new MemoryStorageRoot();
-      this.embeddedStorage.setRoot(this.memoryStorageRoot);
+    this.persistenceStoragePath = persistenceStoragePath;
+    EmbeddedStorageManager storage = null;
+    MemoryStorageRoot root = null;
+    if (persistenceStoragePath != null && !persistenceStoragePath.isEmpty()) {
+      storage =
+          EmbeddedStorage.Foundation(Paths.get(this.persistenceStoragePath))
+              .onConnectionFoundation(
+                  c -> {
+                    c.setFieldEvaluatorPersistable(TRANSIENT_FIELD_EVALUATOR);
+                  })
+              .createEmbeddedStorageManager();
+      storage.start();
+      if (storage.root() == null) {
+        LOG.info("Creating new data storage");
+        root = new MemoryStorageRoot();
+        storage.setRoot(root);
+      } else {
+        LOG.info("Loading existing data from persistence storage");
+        root = (MemoryStorageRoot) storage.root();
+      }
     } else {
-      LOG.info("Loading existing data from persistence storage");
-      this.memoryStorageRoot = (MemoryStorageRoot) this.embeddedStorage.root();
+      root = new MemoryStorageRoot();
```

Comment:
On L127 this explicitly sets the storage.setRoot(root);, but not in this branch? Both are creating a new MemoryStorageRoot() so I don't see where the functional difference comes from.

Reply:
Also, if this path is taken, the storage variable remains null? That would cause NPE in stop().

Reply:
This path avoids persisting data to disk. It's used for testing where persistence is disabled.
The `stop()` method has a null check.